platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/net/ConnectivityReceiver.java

```diff
@@ -85,12 +88,37 @@ public void deactivate() {
    */
   @Override
   public void onReceive(@NonNull Context context, Intent intent) {
-    boolean connected = isConnected(context);
-    Logger.v(TAG, String.format("Connected: %s", connected));
+    if (connected != null) {
+      // Connectivity state overridden by app
+      return;
+    }
+
+    notifyListeners(isNetworkActive());
+  }
+
+  /**
+   * Overwrites system connectivity state. To set, use {@link com.mapbox.mapboxsdk.Mapbox#setConnected(Boolean)}.
+   *
+   * @param connected flag to determine the connectivity state, true for connected, false for
+   *                  disconnected, and null for ConnectivityManager to determine.
+   */
+  public void setConnected(Boolean connected) {
+    this.connected = connected;
+
+    boolean state;
+    if (connected != null) {
+      state = connected;
+    } else {
+      state = isNetworkActive();
+    }
+    notifyListeners(state);
+  }
 
+  private void notifyListeners(boolean isConnected) {
+    Logger.v(TAG, String.format("Connected: %s", isConnected));
```

Comment:
Is it necessary to log this out with every connection change? If yes, can we put the string in a constant?